### ISSUE TITLE
Multiple problems

### ISSUE THREAD

**Issue by OgreTransporter** [jenkinsci/depgraph-view-plugin#30]
The plugin is really useful to see what dependencies exist between projects and then determine an optimal build order. Unfortunately, I am experiencing problems with the menu on the left side of the screen, where you can see two links to the graph. Both lead to the same page. When I display the dependencies I am automatically logged out. It is also no longer possible to click on single projects in the graph.

Jenkins: 2.269
Plugin: 1.0.4

**Comment by dmzobel** [comment]
We came across the same issue and were able to resolve it by clicking the "Keep me signed in" checkbox when logging into the cluster. Still unsure about the root cause though.

**Comment by guidograzioli** [comment]
Thanks for reporting; I am having trouble trying to reproduce this behavior, can you provide more details?

**Comment by OgreTransporter** [comment]
> Thanks for reporting; I am having trouble trying to reproduce this behavior, can you provide more details?

1. i click on the menu item in the main menu for the dependency graph.
2. i can see the graphviz graph
3. i can't click on any project in the graph
4. when I click on any item in the menu or reload the page, I am redirected to the login page.

Is it maybe because I have too many projects (146)?

Jenkins: 2.271

**Comment by guidograzioli** [comment]
> Is it maybe because I have too many projects (146)?

No, at least not on  2.263.1, I have way more on there; I will need to repeat testing on 2.271. Btw, have you got any html sanitizer plugin enabled? I found the labels (alttext) in the graph html map links are broken. Any related logline in the jenkins log?

**Comment by guidograzioli** [comment]
I was able to reproduce the double link in the menu, it happens only for freestyle projects, and it is collateral effect of bug #23 :(

**Comment by OgreTransporter** [comment]
I have only freestyle projects. The plugin didn't work for me with version 2.269 either. Here is my log of once start Jenkins and click on the dependency graph.
```
Running from: D:\Jenkins\jenkins.war
webroot: EnvVars.masterEnvVars.get("JENKINS_HOME")
2020-12-18 12:47:54.022+0000 [id=1]     INFO    org.eclipse.jetty.util.log.Log#initialized: Logging initialized @333ms to org.eclipse.jetty.util.log.JavaUtilLog
2020-12-18 12:47:54.118+0000 [id=1]     INFO    winstone.Logger#logInternal: Beginning extraction from war file
2020-12-18 12:47:55.659+0000 [id=1]     WARNING o.e.j.s.handler.ContextHandler#setContextPath: Empty contextPath
2020-12-18 12:47:55.701+0000 [id=1]     INFO    org.eclipse.jetty.server.Server#doStart: jetty-9.4.33.v20201020; built: 2020-10-20T23:39:24.803Z; git: 1be68755656cef678b79a2ef1c2ebbca99e25420; jvm 1.8.0_241-b07
2020-12-18 12:47:57.918+0000 [id=1]     INFO    o.e.j.w.StandardDescriptorProcessor#visitServlet: NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
2020-12-18 12:47:57.948+0000 [id=1]     INFO    o.e.j.s.s.DefaultSessionIdManager#doStart: DefaultSessionIdManager workerName=node0
2020-12-18 12:47:57.949+0000 [id=1]     INFO    o.e.j.s.s.DefaultSessionIdManager#doStart: No SessionScavenger set, using defaults
2020-12-18 12:47:57.951+0000 [id=1]     INFO    o.e.j.server.session.HouseKeeper#startScavenging: node0 Scavenging every 600000ms
2020-12-18 12:47:58.288+0000 [id=1]     INFO    hudson.WebAppMain#contextInitialized: Jenkins home directory: D:\Jenkins found at: EnvVars.masterEnvVars.get("JENKINS_HOME")
2020-12-18 12:47:58.357+0000 [id=1]     INFO    o.e.j.s.handler.ContextHandler#doStart: Started w.@285f09de{Jenkins v2.271,/,file:///D:/Jenkins/war/,AVAILABLE}{D:\Jenkins\war}
2020-12-18 12:47:58.375+0000 [id=1]     INFO    o.e.j.server.AbstractConnector#doStart: Started ServerConnector@446a1e84{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2020-12-18 12:47:58.375+0000 [id=1]     INFO    org.eclipse.jetty.server.Server#doStart: Started @4686ms
2020-12-18 12:47:58.377+0000 [id=24]    INFO    winstone.Logger#logInternal: Winstone Servlet Engine running: controlPort=disabled
2020-12-18 12:47:59.248+0000 [id=30]    INFO    jenkins.InitReactorRunner$1#onAttained: Started initialization
2020-12-18 12:48:00.187+0000 [id=41]    WARNING hudson.ClassicPluginStrategy#createPluginWrapper: Deprecated unpacked classes directory found in D:\Jenkins\plugins\pyenv\WEB-INF\classes
2020-12-18 12:48:00.243+0000 [id=40]    INFO    jenkins.InitReactorRunner$1#onAttained: Listed all plugins
2020-12-18 12:48:00.457+0000 [id=31]    INFO    o.j.h.p.backup.BackupPluginImpl#loadConfiguration: Loading configuration...
2020-12-18 12:48:00.459+0000 [id=31]    SEVERE  o.j.h.p.backup.BackupPluginImpl#start: Error loading configuration file : D:\Jenkins\backup.xml
2020-12-18 12:48:00.594+0000 [id=47]    INFO    ruby.RubyRuntimePlugin#start: Injecting JRuby into XStream
Trying to load models from D:\Jenkins\plugins\pyenv\WEB-INF\classes\models
Loading D:\Jenkins\plugins\pyenv\WEB-INF\classes\models/pyenv_wrapper.rb
2020-12-18 12:48:04.472+0000 [id=50]    INFO    h.p.virtualbox.VirtualBoxPlugin#start: Starting VirtualBoxPlugin
2020-12-18 12:48:09.698+0000 [id=35]    WARNING o.j.r.u.AnonymousClassWarnings#warn: Attempt to (de-)serialize anonymous class hudson.plugins.disk_usage.DiskUsageProjectActionFactory$DescriptorImpl$1 in file:/D:/Jenkins/plugins/disk-usage/WEB-INF/lib/disk-usage.jar; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
2020-12-18 12:48:09.708+0000 [id=35]    WARNING o.j.r.u.AnonymousClassWarnings#warn: Attempt to (de-)serialize anonymous class hudson.plugins.disk_usage.DiskUsageOvearallGraphGenerator$DiskUsageRecord$1 in file:/D:/Jenkins/plugins/disk-usage/WEB-INF/lib/disk-usage.jar; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
2020-12-18 12:48:10.393+0000 [id=35]    INFO    jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
2020-12-18 12:48:10.395+0000 [id=34]    INFO    c.m.xotradov.SimpleQueuePlugin#postInitialize: Removing BuildQueueWidget
2020-12-18 12:48:10.569+0000 [id=33]    INFO    jenkins.InitReactorRunner$1#onAttained: Started all plugins
2020-12-18 12:48:12.049+0000 [id=29]    INFO    jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
2020-12-18 12:48:12.051+0000 [id=31]    WARNING hudson.plugins.cmake.CmakeTool#tryMigrateLegacy: Could not migrate CMake installation 'CMake' to a format compatible with the zip-installer. Please remove the command name in 'D:\dev\Tools\cmake-3.19.2-win64-x64\bin' on the global tool configuration page.
2020-12-18 12:48:12.117+0000 [id=49]    INFO    jenkins.InitReactorRunner$1#onAttained: System config loaded
2020-12-18 12:48:12.443+0000 [id=69]    INFO    j.p.t.telegram.TelegramBot#initializeProxy: Proxy successfully initialized
2020-12-18 12:48:12.448+0000 [id=69]    INFO    j.p.t.telegram.TelegramBotRunner#lambda$new$0: Bot was created
2020-12-18 12:48:13.532+0000 [id=51]    INFO    jenkins.InitReactorRunner$1#onAttained: System config adapted
2020-12-18 12:48:13.655+0000 [id=38]    INFO    jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2020-12-18 12:48:13.657+0000 [id=38]    INFO    jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2020-12-18 12:48:13.662+0000 [id=31]    INFO    c.a.s.j.p.d.cache.BuildCache#setupWorkerThread: Setting up worker thread pool
2020-12-18 12:48:13.665+0000 [id=31]    INFO    c.a.s.j.p.d.cache.BuildCache#setupGarbageCollector: Setting up GC scheduling
2020-12-18 12:48:13.665+0000 [id=73]    INFO    c.a.s.j.p.d.cache.BuildCache#reloadCache: Building downstream build cache...
2020-12-18 12:48:13.683+0000 [id=75]    INFO    hudson.model.AsyncPeriodicWork#lambda$doRun$0: Started Download metadata
2020-12-18 12:48:13.702+0000 [id=75]    INFO    hudson.model.AsyncPeriodicWork#lambda$doRun$0: Finished Download metadata. 18 ms
2020-12-18 12:48:13.867+0000 [id=34]    INFO    jenkins.InitReactorRunner$1#onAttained: Completed initialization
2020-12-18 12:48:15.146+0000 [id=73]    INFO    c.a.s.j.p.d.cache.BuildCache#reloadCache: Building downstream build cache completed!
2020-12-18 12:48:15.973+0000 [id=23]    INFO    hudson.WebAppMain$3#run: Jenkins is fully up and running

```

**Comment by guidograzioli** [comment]
Unfortunately nothing relevant in your logs; I pushed a new version with a couple of minor fixes, can you check if the situation improved?

**Comment by guidograzioli** [comment]
I was able to reproduce on Jenkins 2.271; by clicking "Keep me signed in" as proposed by @dmzobel the problem disappears. 

It may be related to some newer jenkins changes, as it happens for me for other XHR requests as well (ie. build history checks).

Clicking in the graph is done by downloading an html area map via json after the graph image is generated; when logged in without asking for being kept signed in, the request returns 403 not authorized.

Closing for now: left side menu double link is fixed in 1.0.5; clicking in the graph projects works with workaround ("keep me signed in" checked).
